### PR TITLE
configurable template for new notes

### PR DESCRIPTION
**Pull request by qbikez**
A minimal solution for #48. 
New note template is configurable via `vscodeMarkdownNotes.newNoteTemplate` and defaults to old template, which was previously hard-coded.

### CODE REVIEW

File: src/test/jest/extension.test.ts [hunk 1]
```diff
@@ -137,3 +137,45 @@ test('Note.tagSet', () => {
...
+  it('handles escaped newlines', () => {
+    const template = "# Title\\n\\nContent";
+    
+    const content = newNote(template, 'nevermind');
+
+    expect(content).toBe('# Title\n\nContent');
+  });
+
+  it('handles timestamp', () => {
+    const template = "# Title\n\nCreated: ${timestamp}\n Last modified: ${timestamp}";
```
Comment (kortina) [inline src/test/jest/extension.test.ts:169]: Nit: kind of odd / slightly misleading to have "Last modified" in this example / spec

File: package.json [hunk 2]
```diff
@@ -88,6 +88,11 @@
           "type": "boolean",
           "default": true,
           "description": "By default, when invoking `editor.action.revealDefinition` on `[[note.md]]` if `note.md` does not exist in workspace, create it. NB: Works only when `vscodeMarkdownNotes.workspaceFilenameConvention` = 'uniqueFilenames'."
+        },
+        "vscodeMarkdownNotes.newNoteTemplate": {
+          "type": "string",
+          "default": "# ${noteName}\n\n",
+          "description": "template for auto-created note files. Available tokens: ${noteName}, ${timestamp}"
```
Comment (kortina) [inline package.json:95]: (1) nit: capitalize the "t" in template in the first word of the description
(2) add an example of the timestamp at the end of the description maybe? " (2020-07-09T05:29:00.541Z)."

File: src/test/jest/extension.test.ts (hunk 1 above)
Comment (qbikez) [reply src/test/jest/extension.test.ts]: ok, removed "last modified".

File: package.json (hunk 2 above)
Comment (qbikez) [reply package.json]: Done.

File: src/NoteWorkspace.ts [hunk 3]
```diff
@@ -1,6 +1,7 @@
 import * as vscode from 'vscode';
 import { basename, dirname, join, normalize, relative, resolve } from 'path';
 import { existsSync, readFile, writeFileSync } from 'fs';
+import { config } from 'process';
```
Comment (kortina) [inline src/NoteWorkspace.ts:4]: I am going to merge but then delete this line. Seems like an unnecessary import.

**Pull request merged**